### PR TITLE
TOFU v2 evals: finish new design of eval pipeline, add documentation. 

### PR DESCRIPTION
**Pull request by Dornavineeth**
# Done
- [x] Refactor Eval to re-use probability handler for all metrics such as Q_A_Prob, Q_A_PARA_Prob
- [x] Added `handler` for each config package
- [x] Added pre-compute option to the Metric 

# TODO
- [ ] Standardize output of each metric
- [ ] Implement all TOFU Metrics
- [ ] Integrate evalutation code in Trainers
- [ ] Add FSDP feature

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,20 +44,80 @@ pip install -r requirements.txt
...
 
-
 ## Tasks at hand
 
+For detailed documentation please find the links below.
 | Tasks | Link                                            |
 |-----------|------------------------------------------------|
 | Unlearn and implement new unlearning methods       | [here](docs/unlearning.md)     |
 | Evaluate and implement new metrics for unlearning  | [here](docs/evaluation.md)     |
 | Train models for developing new benchmark.              | [here](docs/finetune.md)       |
```
Comment (molereddy) [inline README.md:54]: Just say: finetune models to learn from a dataset

File: README.md [hunk 2]
```diff
@@ -44,20 +44,80 @@ pip install -r requirements.txt
...
+- **data.retain=TOFU_QA_RETAIN90**: Sets the retain dataset for [QADataset](../src/data/tofu.py) with config [TOFU_QA_RETAIN90.yaml](../configs/data/datasets/TOFU_QA_RETAIN90.yaml) for unlearning.
+- **data.eval=TOFU_QA_FORGET10_P**: Sets the evaluation dataset for [QADataset](../src/data/tofu.py) with config [TOFU_QA_FORGET10_P.yaml](../configs/data/datasets/TOFU_QA_FORGET10_P.yaml) for unlearning.
+
+
+### Evaluate and implement new metrics for unlearning
+
+To run TOFU benchmark
+```bash
+python src/eval.py \
+model=Llama-3.1-8B-Instruct \ # Model to evaluate
```
Comment (molereddy) [inline README.md:89]: where do we give path to model we are evaluating?

File: configs/eval/tofu.yaml [hunk 3]
```diff
@@ -12,7 +12,11 @@ defaults: # include all defined metrics files
     - Q_A_PERT_ROUGE
     - BIO_Prob
     # - BIO_ROUGE
+    - Q_A_PARA_Prob
+    - Truth_Ratio
 
+handler: TOFUEvaluator
 device: cuda
```
Comment (molereddy) [inline configs/eval/tofu.yaml:19]: is this line to ensure eval runs on a single GPU? are we enforcing single-GPU evals?

File: configs/eval/tofu.yaml
Comment (molereddy) [inline configs/eval/tofu.yaml:1]: is the tofu_metrics key accessed anywhere or is it just to organize things? if the latter, maybe better to clarify things

File: README.md (hunk 2 above)
Comment (molereddy) [reply README.md:89]: add commented sample path

**Review by molereddy — COMMENTED** [review]
just elaborate on the tofu_metric v/s package thing in config and add sample path (commented) in quickstart docs

File: configs/eval/tofu.yaml
Comment (molereddy) [reply configs/eval/tofu.yaml:1]: if tofu_metrics is just to group things in config, better use a comment than an actual key
can you remove the tofu_metrics key and replace with some generic placeholder put it in comments?

File: README.md [hunk 4]
```diff
@@ -44,20 +44,81 @@ pip install -r requirements.txt
...
+eval=tofu # evaluation config to run (e.g. tofu benchmark)
```
Comment (molereddy) [inline README.md:91]: if you run copy-paste and run this kind of command directly the # comments gets in the way and gives error
this is used in many places, leading to errors
decide a better way of doing this for the entire documentation

File: README.md (hunk 2 above)
Comment (Dornavineeth) [reply README.md:90]: Fixed the inline comments issue. For now lets leave a place holder for the path as you did. @molereddy

File: configs/eval/tofu.yaml (hunk 3 above)
Comment (Dornavineeth) [reply configs/eval/tofu.yaml:19]: Yea. For now our code assumes that everything runs on machine. Will take of using DDP and FSDP later. 
Will add this as todo in merge request.

**Pull request merged**